app/github_repo_fetcher.rb

```diff
@@ -3,27 +3,16 @@
 class GitHubRepoFetcher
   include Singleton
 
-  # Fetch a repo from GitHub.
-  #
-  # If the app_name is just a repo name we'll assume that it's in the alphagov
-  # org. Because 99% of the repos will be from that org, we can download and
-  # cache all alphagov-repos and save on API calls.
+  # Fetch a repo from GitHub
   def repo(app_name)
-    if app_name =~ %r{/}
-      # Not on alphagov, make a separate call to the API. Cache it for
-      # development speed.
-      @cache ||= {}
-      @cache[app_name] ||= client.repo(app_name)
-    else
-      all_alphagov_repos.find { |repo| repo.name == app_name } || raise("alphagov/#{app_name} not found")
-    end
+    all_alphagov_repos.find { |repo| repo.name == app_name } || raise("alphagov/#{app_name} not found")
   end
 
   # Fetch a README for an alphagov application and cache it.
   # Note that it is cached as pure markdown and requires further processing.
   def readme(app_name)
     CACHE.fetch("alphagov/#{app_name} README", expires_in: 1.hour) do
-      Base64.decode64(client.readme("alphagov/#{app_name}").content)
+      HTTP.get("https://raw.githubusercontent.com/alphagov/#{app_name}/master/README.md")
```

Comment:
If we ever start changing some default branch names to non-`master` but not others, this will break. But we can cross that bridge when we come to it...

Reply:
Argh, that's a very good point! I'd be tempted to hold off merging this for that alone 😕 unless we plan on changing branch defaults en masse rather than in dribs and drabs.

Reply:
The GitHub API does return `default_branch` for a repo, so we could reintroduce some code and grab it programatically to mean we have to do less manual work when things change, and we can make this improvement to README grabbing now.

Reply:
Thanks - looks like we already have access to that information, as we cache all the repo info up front, so it's a fairly trivial code change. Unfortunately the tests are another matter entirely 😓 

This is what I have so far but I'm hoping to refactor tomorrow:
https://github.com/alphagov/govuk-developer-docs/pull/2743/commits/3f12972ab4ce0152322b7e073fd3ed3db61c73d7

Reply:
Have had another look at this today and whilst it's a bit of a smell in places, I think it's mergeable. Think it would take quite a rewrite to make it cleaner, but with the repo branch name changes happening over the next few weeks I'd sooner just remove my last commit and wait for everything to be on `main` than attempt a rewrite now.

Will request a review from you and see what you think 😄